### ISSUE TITLE
Implement API system based on DRF token auth

### ISSUE THREAD

**Issue by jwalgran** [open-apparel-registry/open-apparel-registry#75]
## Overview

Allow authenticated requests to be made with a token instead of a username and password.

### Describe the solution you'd like

 - Add an authenticated DRF endpoint for creating and deleting tokens associated with a User.
  - Update the frontend to use the new endpoint. The old app uses a hardcoded API key, so we don't have to worry about breaking compatibility between the updated front end and the old backend. 
    - Users can now have multiple API keys. Modify the frontend to allow for this.
  - To exercise token auth, add a secured endpoint that returns all FacilityList rows associated with the provided API key / token (the `getLists` route from the old application).

### Additional Context

Depends on #70

**Comment by kellyi** [comment]
> Users can now have multiple API keys. Modify the frontend to allow for this.

After discussing this with @jwalgran I'm going to create a new issue for the multiple tokens part and just set up the API to afford having one token per user. I'm updating the UI in #110 to potentially enable having multiple tokens, but the updates to the API for multiple tokens will be a little more involved.

DRF itself intentionally sticks to a one-token-per-user model. There are a couple libraries which enable having multiple tokens, and it's also potentially possible to create one without a library. I'll make a note to that effect on the new issue.
